premis_event_service/views.py

```diff
@@ -95,91 +96,104 @@ def humanEvent(request, identifier=None):
     )
 
 
-def paginate_entries(request, entries, num_per_page=20):
-    """
-    paginates a set of entries (set of model objects)
-    """
-
-    # create paginator from result set
-    paginator = Paginator(entries, num_per_page)
-    paginated_entries = None
-    # try to resolve the current page
-    try:
-        page = int(request.GET.get('page', '1'))
-    except ValueError:
-        page = 1
-    try:
-        paginated_entries = paginator.page(page)
-    except (EmptyPage, InvalidPage):
-        paginated_entries = paginator.page(paginator.num_pages)
-    # send back the paginated entries
-    return paginated_entries
+def last_page_ordinal(query_set, per_page=20):
+    qa = query_set.order_by('ordinal')[0:20]
+    return qa.last().ordinal
+
+
+def paginate_events(valid, request, per_page=20):
+    total_events = None
+    events = None
+    page = int(request.GET.get('page', 1))
+    offset = None
+    unfiltered = True
+    last_page_ord = per_page+1
+    if any([v for k, v in valid.items() if k != 'min_ordinal']):
+        events = (Event.objects.search(**valid)
+                  .prefetch_related('linking_objects'))
+        total_events = events.count()
+        offset = (page-1)*per_page
+        last_page_ord = last_page_ordinal(events)
+        if 'min_ordinal' in valid and valid['min_ordinal']:
+            events = events.filter(ordinal__lte=valid['min_ordinal'])[:20]
+        else:
+            events = events[offset:offset+per_page]
+        unfiltered = False
+    else:
+        events = Event.objects.searchunfilt(request.GET.get('min_ordinal'))
+        total_events = Event.objects.all().count()
+        events = events.prefetch_related('linking_objects')[0:per_page]
+    page_max_ord = 0
+    page_min_ord = 0
+    if events:
+        page_max_ord = max([e.ordinal for e in events])
+        page_min_ord = min([e.ordinal for e in events])
+    if page > math.ceil(float(total_events)/per_page):
+        if page > 1:
+            raise EmptyPage()
+    max_page = int(math.ceil(total_events/float(per_page)))
```

Comment:
I'm probably missing something, but those two conditionals seem to be identical other than one returns a float and the other an integer. But they both should behave the same, given how they're used here. And that special case of max_page being 0 seems to be covered by the `if page > 1` conditional.